src/precice/config/ParticipantConfiguration.cpp

```diff
@@ -495,8 +494,9 @@ void ParticipantConfiguration::finishParticipantConfiguration(
   _mappingConfig->resetMappings();
 
   // Set participant data for data contexts
-  for (impl::DataContext &dataContext : participant->writeDataContexts()) {
-    int fromMeshID = dataContext.getMeshID();
+  for (const auto &item : participant->writeDataContexts()) {
+    impl::WriteDataContext &dataContext = *(item.second.get());
```

Comment:
See comment in SolverInterfaceImpl.cpp

Reply:
See 9fab184e72835431b1197706c0f7e39419e32413.